### PR TITLE
Add Travis support

### PR DESCRIPTION
**Pull request by gsmet**
Hi!

As mentioned on the list, I think we could give Travis a try for building the PRs.

Note that what is interesting with Travis is that it can also run for our individual forks so we can get our branches tested by the CI before creating the PR.

The .travis.yml build the tests for 3 different RDBMS, maybe we could just build it for PostgreSQL if we only use Travis for the PR tests?

Thoughts?

### CODE REVIEW

**Comment by gsmet** [comment]
Note that we can commit the .travis.yml file without enabling Travis on the main project. But contributors could easily enable CI for their forks!

**Comment by gsmet** [comment]
@Sanne I thought about https://github.com/hibernate/hibernate-search/pull/1008 and I think it would be cool to add a checkstyle:check at the beginning of the Travis build to fail fast if something is wrong with checkstyle.

I was thinking about adding that, only test for h2 for now and merge this PR. Anything against this plan?

**Comment by Sanne** [comment]
I think we could merge this; I'm not sure yet if I'll like Travis but these changes look harmless so why not..

Could you create a JIRA for this and refer to it in the commit log?

**Comment by Sanne** [comment]
> I thought about #1008 and I think it would be cool to add a checkstyle:check at the beginning of the Travis build to fail fast if something is wrong with checkstyle.

I think after our checkstyle fixes from today this should not be necessary anymore, right?
AFAIK we run the check as part of each build now.

**Comment by gsmet** [comment]
What I like with doing a checkstyle:check before is that we fail fast in case of style errors as we check all the code source before running the tests.

**Comment by Sanne** [comment]
ok.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  - DB=pgsql
+  - DB=mariadb
+install:
+  - mvn -v
+  - mvn -s settings-example.xml -B -q clean compile -DskipTests=true
+before_script: 
+  - if [[ "$DB" == "pgsql" ]]; then psql -U postgres -f 'travis/pgsql.sql';fi 
+  - if [[ "$DB" == "mariadb" ]]; then mysql -u root < 'travis/mariadb.sql'; fi
+script:
+  - if [[ "$DB" == "h2" ]]; then mvn -s settings-example.xml -Ph2 test;fi
```
Comment (gunnarmorling) [inline .travis.yml]: How comes you do the tests after compile? Seems btw. it'd miss the integration tests that way?

File: backends/jgroups/src/test/java/org/hibernate/search/backend/jgroups/impl/SyncJGroupsBackendTest.java [hunk 2]
```diff
@@ -114,10 +114,15 @@ public void testSynchAsConfigured() {
 		}
 		catch (SearchException se) {
 			//Expected: we're inducing the RPC into NPE
-			Throwable cause = se.getCause().getCause().getCause();
-			Assert.assertTrue( "Cause was not a NullPointerException but a " + cause, cause instanceof NullPointerException );
-			Assert.assertEquals( "Simulated Failure", cause.getMessage() );
-			npeTriggered = true;
+			try {
```
Comment (gunnarmorling) [inline backends/jgroups/src/test/java/org/hibernate/search/backend/jgroups/impl/SyncJGroupsBackendTest.java]: How is this related to the Travis set-up?

**Comment by gunnarmorling** [comment]
+1 for merging and getting it off our plates :)

File: backends/jgroups/src/test/java/org/hibernate/search/backend/jgroups/impl/SyncJGroupsBackendTest.java (hunk 2 above)
Comment (gsmet) [inline backends/jgroups/src/test/java/org/hibernate/search/backend/jgroups/impl/SyncJGroupsBackendTest.java]: Well, it's related because I needed the exception to understand what was going on and fix it :)

Comment (Sanne) [inline backends/jgroups/src/test/java/org/hibernate/search/backend/jgroups/impl/SyncJGroupsBackendTest.java]: Yea, I know were you're coming from with this fix.. but I think that @gunnarmorling was suggesting that such improvements should live in a separate commit at least, sometimes in their own JIRA.

For example in this case I don't think this is the best improvement for that NPE but asking to improve on that would delay integration of the other improvements in this PR too.

**Comment by gsmet** [comment]
OK, so I fixed everything you reported and added comments to the .travis.yml file.

The bad news is that the elasticsearch integration tests don't seem to be stable enough. I had SocketTimeoutExceptions.

I committed higher timeouts to see how it goes.

**Comment by Sanne** [comment]
> I committed higher timeouts to see how it goes.

What do you prefer,shall we merge this or are you going to try the new timeouts out?

**Comment by gsmet** [comment]
It's better with the higher timeout but I still have 2 tests failing:
- one because the test run before the cluster was green: so we really need what you talked about in your last email;
- as for the other, it looks like a real failure but it might be a timing problem too:
```
testScenario(org.hibernate.search.test.engine.optimizations.CollectionUpdateEventsSecondTest)  Time elapsed: 0.241 sec  <<< FAILURE!
java.lang.AssertionError: expected:<0> but was:<1>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:743)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:555)
	at org.junit.Assert.assertEquals(Assert.java:542)
	at org.hibernate.search.test.engine.optimizations.CollectionUpdateEventsSecondTest.assertOperationsPerformed(CollectionUpdateEventsSecondTest.java:102)
```

**Comment by Sanne** [comment]
I rebased and merged the first 3 commits as they were nice and independent;

The 4th commit "Try to use a higher timeout for Elasticsearch" sounds like an experiment and doesn't have a JIRA yet.
Let me know what you think we should do with that commit? Or shall we close this PR?

**Comment by gsmet** [comment]
Gunnar sets the timeout higher in https://github.com/hibernate/hibernate-search/pull/1023 so I'm closing this one for now.
